[DataPipe] Unifying API - removing nesting_level argument from MapperIterDataPipe

NivekT opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69392
* #69393
* #69391
* __->__ #69390

As part of the efforts to unify the APIs across different data backends (e.g. TorchData, TorchArrow), we are making changes to different DataPipes' APIs. In this PR, we are removing the input argument `nesting_level` from `MapperIterDataPipe`.

cc @VitalyFedyunin @ejguan @NivekT

Differential Revision: [D32849465](https://our.internmc.facebook.com/intern/diff/D32849465)

NivekT committed:
[DataPipe] unifying API - removing nesting_level argument from MapIterDataPipe

[ghstack-poisoned]

ejguan reviewed (approved):
LGTM, thanks. Comment is in the other PR https://github.com/pytorch/pytorch/pull/69392#discussion_r762304026.

NivekT committed:
Update on "[DataPipe] Unifying API - removing nesting_level argument from MapperIterDataPipe"


As part of the efforts to unify the APIs across different data backends (e.g. TorchData, TorchArrow), we are making changes to different DataPipes' APIs. In this PR, we are removing the input argument `nesting_level` from `MapperIterDataPipe`.

cc @VitalyFedyunin ejguan @NivekT

[ghstack-poisoned]

NivekT commented:
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32849465).
